akka-persistence-testkit/src/main/scala/akka/persistence/testkit/javadsl/EventSourcedBehaviorTestKit.scala

```diff
@@ -0,0 +1,242 @@
+     * or [[CommandResult.eventOfType]].
+     */
+    def events: JList[Event] =
+      delegate.events.asJava
+
+    /**
+     * `true` if no events were emitted by the command.
+     */
+    def hasNoEvents: Boolean =
+      delegate.hasNoEvents
+
+    /**
+     * The first event. It will throw `AssertionError` if there is no event.
+     */
+    def event: Event =
+      delegate.event
+
+    /**
+     * The first event as a given expected type. It will throw `AssertionError` if there is no event or
+     * if the event is of a different type.
+     */
+    def eventOfType[E <: Event](eventClass: Class[E]): E =
+      delegate.eventOfType(ClassTag[E](eventClass))
+
+    /**
+     * The state after applying the events.
+     */
+    def state: State =
+      delegate.state
+
+    /**
+     * The state as a given expected type. It will throw `AssertionError` if the state is of a different type.
+     */
+    def stateOfType[S <: State](stateClass: Class[S]): S =
+      delegate.stateOfType(ClassTag[S](stateClass))
+  }
+
+  /**
+   * The result of running a command with a `ActorRef<R> replyTo`, i.e. the `runCommand` with a
+   * `Function<ActorRef<R>, Command>` parameter.
+   */
+  final class CommandResultWithReply[Command, Event, State, Reply](
+      delegate: scaladsl.EventSourcedBehaviorTestKit.CommandResultWithReply[Command, Event, State, Reply])
+      extends CommandResult[Command, Event, State](delegate) {
+
+    /**
+     * The reply. It will throw `AssertionError` if there was no reply.
+     */
+    def reply: Reply =
+      delegate.reply
+
+    /**
+     * The reply as a given expected type.  It will throw `AssertionError` if there is no reply or
+     * if the reply is of a different type.
+     */
+    def replyOfType[R <: Reply](replyClass: Class[R]): R =
+      delegate.replyOfType(ClassTag[R](replyClass))
+  }
+
+  /**
+   * The result of restarting the behavior.
+   */
+  final class RestartResult[State](delegate: scaladsl.EventSourcedBehaviorTestKit.RestartResult[State]) {
+
+    /**
+     * The state after recovery.
+     */
+    def state: State =
+      delegate.state
+  }
+
+}
+
+@ApiMayChange
+final class EventSourcedBehaviorTestKit[Command, Event, State](
+    delegate: scaladsl.EventSourcedBehaviorTestKit[Command, Event, State]) {
+
+  import EventSourcedBehaviorTestKit._
+
+  private val _persistenceTestKit = new PersistenceTestKit(delegate.persistenceTestKit)
+
+  /**
+   * Customization of which serialization checks that are performed.
+   * By default serialization of commands, events and state are verified, but equality of the
+   * serialization roundtrip is not checked. `equals` must be implemented (or using `case class`) in the
+   * commands, events and state if `verifyEquality` is enabled.
+   */
+  def setSerializationSettings(settings: SerializationSettings): EventSourcedBehaviorTestKit[Command, Event, State] = {
```

Comment:
The EventSourcedBehaviorTestKit instance isn't immutable, it holds a stateful actor of the behavior.

The reason I made it a setter instead of constructor parameter was that then it's possible to change for different commands, but that seems to be overkill. Then new testkit can be created.

I'll change it to constructor parameter.